sql/4_2_2-to-4_3_1_upgrade.sql

```diff
@@ -71,4 +71,129 @@
 --  #IfNotListOccupation
 --    Custom function for creating Occupation List
 
+--  #IfTableEngine
+--    desc:      Execute SQL if the table has been created with given engine specified.
+--    arguments: table_name engine
+--    behavior:  Use when engine conversion requires more than one ALTER TABLE
 
+--  #IfInnoDBMigrationNeeded
+--    desc: find all MyISAM tables and convert them to InnoDB.
+--    arguments: none
+--    behavior: can take a long time.
+
+
+--
+-- The following tables contains TEXT NOT NULL DEFAULT "" declaration which is not compatible with InnoDB.
+-- We omit NOT NULL DEFAULT "" declaration.
+--
+
+#IfTableEngine history_data MyISAM
+-- remove NOT NULL DEFAULT "" declaration from TEXT fields.
+ALTER TABLE `history_data` MODIFY `exams` TEXT, MODIFY userarea11 TEXT, MODIFY userarea12 TEXT NOT NULL, ENGINE=InnoDB;
+#EndIf
+
+#IfTableEngine lang_custom MyISAM
+-- remove NOT NULL DEFAULT "" declaration from MEDIUMTEXT fields.
+ALTER TABLE `lang_custom` MODIFY `constant_name` mediumtext, MODIFY `definition` mediumtext, ENGINE=InnoDB;
+#EndIf
+
+#IfTableEngine layout_options MyISAM
+-- remove NOT NULL DEFAULT "" declaration from TEXT field.
+ALTER TABLE `layout_options` MODIFY `conditions` text COMMENT 'serialized array of skip conditions', ENGINE=InnoDB;
+#EndIf
+
+#IfTableEngine patient_data MyISAM
+-- remove NOT NULL DEFAULT "" declaration from TEXT field.
+ALTER TABLE `patient_data` MODIFY  `billing_note` text, ENGINE=InnoDB;
+#EndIf
+
+#IfTableEngine rule_action_item MyISAM
+-- remove NOT NULL DEFAULT "" declaration from TEXT field.
+ALTER TABLE `rule_action_item` MODIFY `reminder_message` text COMMENT 'Custom message in patient reminder', ENGINE=InnoDB;
+#EndIf
+
+#IfTableEngine procedure_providers MyISAM
+-- remove NOT NULL DEFAULT "" declaration from TEXT field.
+ALTER TABLE `procedure_providers` MODIFY `notes` TEXT, ENGINE=InnoDB;
+#EndIf
+
+#IfTableEngine procedure_questions MyISAM
+-- remove NOT NULL DEFAULT "" declaration from TEXT field.
+ALTER TABLE `procedure_questions` MODIFY `options` text COMMENT 'choices for fldtype S and T', ENGINE=InnoDB;
+#EndIf
+
+
+#IfTableEngine procedure_order MyISAM
+-- remove NOT NULL DEFAULT "" declaration from TEXT field.
+ALTER TABLE `procedure_order` MODIFY `patient_instructions` TEXT, ENGINE=InnoDB;
+#EndIf
+
+-- #IfTableEngine procedure_order_code MyISAM - see below
+-- remove NOT NULL DEFAULT "" declaration from TEXT field.
+-- ALTER TABLE `procedure_order_code` MODIFY `diagnoses` TEXT COMMENT 'diagnoses and maybe other coding (e.g. ICD9:111.11)', ENGINE=InnoDB;
+-- #EndIf
```

Comment:
We grouped tables according to the treatments they need to receive. procedure_order_code table needs modification for  AUTO_INCREMENT problem and for TEXT DEFAULT problem. This section was left commented out to indicate that it's still member of TEXT DEFAULT.